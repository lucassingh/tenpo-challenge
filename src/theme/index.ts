import light from './lightTheme';

export default { light }