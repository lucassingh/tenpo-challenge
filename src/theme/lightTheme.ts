export default {
    colors: {
        primary: "#D4F9F5",
        secondary: "#00BAA4",
        background: "#F2F2F2",
        lightGray: "#F2F2F2",
        light: "#FFFFFF",
        dark: "#1C1C1C"
    }
}