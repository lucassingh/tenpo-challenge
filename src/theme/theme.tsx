import React from "react";
import { ThemeProvider } from "styled-components/native";
import theme from ".";

type Props = {
    children: React.ReactNode
}

export function Theme({children}: Props) {
    return <ThemeProvider theme={theme.light}>
        {children}
    </ThemeProvider>
}