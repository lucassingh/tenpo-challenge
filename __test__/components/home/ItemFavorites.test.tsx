import React from "react";
import 'jest-styled-components'
import styled from 'styled-components/native';
import theme from "../../../src/theme";
import { render } from '@testing-library/react-native'

describe('Test ItemContainer and text in component <ItemFavorites />', () => {

    const ItemContainer = styled.View`
        display: flex;
        flex-direction: column;
        width: 250px;
        height: 150px;
        border: 1px solid ${theme.light.colors.lightGray};
        margin-right: 10px;
        border-radius: 10px;
    `

    it('Should match with snapshot', () => {

        const container = render(<ItemContainer />);

        expect(container).toMatchSnapshot()
    })

    const Title = styled.Text`
        color: ${theme.light.colors.secondary};
        font-weight: 700; 
    `

    it('Should match with snapshot', () => {

        const container = render(<Title />);

        expect(container).toMatchSnapshot()
    })
})