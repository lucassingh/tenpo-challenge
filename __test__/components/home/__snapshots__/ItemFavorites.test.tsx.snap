// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ItemContainer and text in component <ItemFavorites /> Should match with snapshot 1`] = `
<View
  style={
    [
      {
        "borderColor": "#F2F2F2",
        "borderRadius": 10,
        "borderStyle": "solid",
        "borderWidth": 1,
        "display": "flex",
        "flexDirection": "column",
        "height": 150,
        "marginRight": 10,
        "width": 250,
      },
    ]
  }
/>
`;

exports[`Test ItemContainer and text in component <ItemFavorites /> Should match with snapshot 2`] = `
<Text
  style={
    [
      {
        "color": "#00BAA4",
        "fontWeight": "700",
      },
    ]
  }
/>
`;
